frontend/src/metabase/lib/react-compat.ts

```diff
@@ -4,14 +4,16 @@ import React from "react";
 import ReactDOM from "react-dom";
 import { type Root, createRoot } from "react-dom/client";
 
+import { isReactVersionLessThanOrEqualTo17 } from "./compat/check-version";
+
 // React 18 and later has the useSyncExternalStore hook.
 export const isReact17OrEarlier = () => !("useSyncExternalStore" in React);
 
 export function renderRoot(
   content: React.JSX.Element,
   element: Element,
 ): Root | undefined {
-  if (isReact17OrEarlier()) {
+  if (isReactVersionLessThanOrEqualTo17(React.version)) {
```

Comment:
Using `React.version` as it seems to be the best way to tell which version the app is on at runtime